core/types/transaction.go

```diff
@@ -84,6 +85,12 @@ func NewContractCreation(nonce uint64, shardID uint32, amount *big.Int, gasLimit
 	return newTransaction(nonce, nil, shardID, amount, gasLimit, gasPrice, data)
 }
 
+// NewStakeTransaction constructs a stake transaction.
+// Currently a stake transaction is sent to a constant beaconchain address and the shard id is 0.
```

Comment:
The second line of comment better be somewhere else where you send the transaction. The stake transaction by itself can be implemented without hack, so no need to put how the stake tx is used in this package.